todo-app/src/TodoDesc.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react'
+import './todo.css'
+
+export default class TodoDesc extends React.Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            updatedToDoDesc: this.props.todoDesc
+        }
+    }
+    handleInputChange = (event) => {
+        this.setState({
+            updatedToDoDesc: event.target.value
+        })
+    }
+    render = () => {
+        let readOnlyOPT = {
+            'readOnly': this.props.isUpdating ? "" : "readOnly"
+        }
+        return (
+            <div className="toDo-Description">
+                <dev  >
+                    <input type="text" name="username" value={this.state.updatedToDoDesc} onChange={this.handleInputChange} {...readOnlyOPT} />
+                </dev>
+                <div><a href="#" className="editToDo" name="Edit" onClick={this.props.controlActions.editToDo} >{this.props.isUpdating ? "Cancel" : "Edit"}</a></div>
```

Comment:
We should not use `href="#"` for `a` tags. We als get a nice warning from React.

<img width="616" alt="screen shot 2017-11-09 at 08 22 17" src="https://user-images.githubusercontent.com/1057324/32593165-28eb72b2-c527-11e7-9b7f-3989dfcac437.png">